File: php/test/Creational/AbstractFactoryTest.php
<?

require_once 'src/Creational/AbstractFactory/BostonConcreteSoupFactory.php';
require_once 'src/Creational/AbstractFactory/HonoluluConcreteSoupFactory.php';

class Creational_AbstractFactoryTest extends PHPUnit_Framework_TestCase
{

	function MakeSoupOfTheDay($concreteSoupFactory)
	{
		$soups = [ $concreteSoupFactory->makeChickenSoup(),
           $concreteSoupFactory->makeClamChowder(),
           $concreteSoupFactory->makeFishChowder(),
           $concreteSoupFactory->makeMinnestrone(),
           $concreteSoupFactory->makePastaFazul(),
           $concreteSoupFactory->makeTofuSoup(),
           $concreteSoupFactory->makeVegetableSoup(),
           $concreteSoupFactory->makeVegetableSoup(),
       ];
		return $soups[3];
	}        

   public function test()
   {
		$concreteSoupFactory = new BostonConcreteSoupFactory();
		$soupOfTheDay = $this->MakeSoupOfTheDay($concreteSoupFactory);
		$this->assertEquals("The Soup of the day " .  $concreteSoupFactory->factoryLocation .  " is " .  $soupOfTheDay->soupName,
			"The Soup of the day Boston is Minnestrone",
			'Testing boston soup of the day');

		$concreteSoupFactory = new HonoluluConcreteSoupFactory();
		$soupOfTheDay = $this->MakeSoupOfTheDay($concreteSoupFactory);
		$this->assertEquals("The Soup of the day " .  $concreteSoupFactory->factoryLocation .  " is " .  $soupOfTheDay->soupName,
			"The Soup of the day Honolulu is Minnestrone",
			'Testing boston soup of the day');
	}
}

File: go/creational/abstractfactory/abstractfactory_test.go
//TestAbstractSoupFactory - Testing the abstract factory

package abstractfactory

import (
	"fmt"
	"testing"
	"time"
	"../../lib/asserts"
	"../common"
)

func TestAbstractFactory (t *testing.T) { 
	
   makeSoupOfTheDay := func (concreteSoupFactory AbstractSoupFactory) common.Soup { 
		dayOfWeek := time.Tuesday //Calendar.TUESDAY
      
		switch( dayOfWeek ) {
			case time.Monday:
				return concreteSoupFactory.MakeChickenSoup()
			case time.Tuesday:
				return concreteSoupFactory.MakeClamChowder()
			case time.Wednesday :
				return concreteSoupFactory.MakeFishChowder()
			case time.Thursday :
				return concreteSoupFactory.MakeMinnestrone()
			case time.Friday :
				return concreteSoupFactory.MakePastaFazul()
			case time.Saturday :
				return concreteSoupFactory.MakeTofuSoup()
			case time.Sunday :
			default:
				return concreteSoupFactory.MakeVegetableSoup()
       }
		return concreteSoupFactory.MakeVegetableSoup()
   }        

	boston := NewBostonConcreteSoupFactory()
	soupOfTheDay := makeSoupOfTheDay(boston)
	
	asserts.Equals( t, "Boston Abstract Factory test",
		"The Soup of the day Boston is QuahogChowder",
		"The Soup of the day " + boston.GetFactoryLocation() + " is " + soupOfTheDay.GetSoupName())

	hono := NewHonoluluConcreteSoupFactory()
	soupOfTheDay = makeSoupOfTheDay(hono)
	asserts.Equals( t, "Honolulu Abstract Factory test",
		"The Soup of the day Honolulu is PacificClamChowder",
		"The Soup of the day " + hono.GetFactoryLocation() + " is " + soupOfTheDay.GetSoupName())
} 
